assets/js/src/listing/search.jsx

```diff
@@ -7,13 +7,17 @@ class ListingSearch extends React.Component {
     super(props);
     this.state = {
       search: '',
+      componentMounted: false,
     };
     this.handleSearch = this.handleSearch.bind(this);
     this.onChange = this.onChange.bind(this);
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.setState({ search: nextProps.search });
+  static getDerivedStateFromProps(prevProps, prevState) {
+    if (!prevState.componentMounted && prevProps.search) {
```

Comment:
This condition passes only the first time that `props.search` has a value. If this prop changes after that, the state will not!
I think using `componentDidMount` and comparing previous and next props would be better to reproduce the same behavior as before.

Reply:
I don't understand what do you mean... I don't see how using `componentDidMount` could help in your scenario. Isn't that only executed once at the beginning of the lifecycle? So if this prop changes after that, the state will not anyway? What am I missing?

Reply:
Sorry, I meant `componentDidUpdate`!